components/create_post/create_post.jsx

```diff
@@ -524,11 +530,26 @@ export default class CreatePost extends React.Component {
         post.create_at = time;
         post.parent_id = this.state.parentId;
 
+        const hookResult = await actions.runMessageWillBePostedHooks(post);
+
+        if (hookResult.error) {
+            this.setState({
+                serverError: hookResult.error,
```

Comment:
Not really a server error.

Reply:
The name is bad, but I didn't want to go through with changing it elsewhere. Ideally, I think we'd just have one "error" state that controls the error message instead of multiple different fields depending on where the error came from, but seeing as there's already `this.state.serverError` and `this.state.textError`, I didn't want to add another one.